web/app/js/components/util/TapUtils.jsx

```diff
@@ -312,18 +312,14 @@ export const directionColumn = d => (
   </Tooltip>
 );
 
-export const srcDstColumn = (d, resourceType, ResourceLink) => {
-
-  let display = {};
-  let labels = {};
+export const extractDisplayName = d => {
+  let display = d.destination || {};
+  let labels = d.destinationLabels || {};
+  return [labels, display];
+};
 
-  if (d.direction === "INBOUND") {
-    display = d.source;
-    labels = d.sourceLabels;
-  } else {
-    display = d.destination;
-    labels = d.destinationLabels;
-  }
```

Comment:
We still need to check the direction of the event here before extracting the name. If this is an inbound event, then we want to use the source, whereas as written you're always using the destination. I think this fixes it:

```diff
diff --git a/web/app/js/components/util/TapUtils.jsx b/web/app/js/components/util/TapUtils.jsx
index d8b60243..3b95cdec 100644
--- a/web/app/js/components/util/TapUtils.jsx
+++ b/web/app/js/components/util/TapUtils.jsx
@@ -313,8 +313,16 @@ export const directionColumn = d => (
 );
 
 export const extractDisplayName = d => {
-  let display = d.destination || {};
-  let labels = d.destinationLabels || {};
+  let display = {};
+  let labels = {};
+
+  if (d.direction === "INBOUND") {
+    display = d.source;
+    labels = d.sourceLabels;
+  } else {
+    display = d.destination;
+    labels = d.destinationLabels;
+  }
   return [labels, display];
 };
 
```

Reply:
Thanks for the catch, my new commit should resolve this.